Changes made for PhaseSystem to work with the Waxing/Waning

WispSS13 opened the pull request:
Changes made on 18/11/2020 to allow us to better use the waxing/waning mechanic with more cards and relics,

WispSS13 committed:
Update ModTheSpire.json

WispSS13 committed:
Changes to PhaseSystem for Waxing/Waning mechanic

Changes made for easier use with cards and relics

strangeglyph commented on src/main/java/org/m14n/moonthespire/PhaseSystem.java:
```diff
@@ -57,6 +69,31 @@ private static void modifyDexterity(int amt) {
+    }
+
+
     static void reset() {
```
Oops, we missed this when we were updating PhaseSystem. Could you reset wasWaxing and wasWaning too, here?

WispSS13 replied:
Sure thing, wasWaxing and wasWaning now reset to False.

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+    public static final String IMG = cardPath("moondrop.png");
+
+    private static final int COST = 2;
+    private static final CardType TYPE = CardType.ATTACK;
```
Should we change the card type alongside the target in onPhaseChange?

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+    private static final CardColor COLOR = MoonCharacter.Enums.MOON_PURPLE;
+    private static final CardRarity RARITY = CardRarity.UNCOMMON;
+    private static final CardTarget TARGET = CardTarget.ENEMY;
+    private static final CardStrings cardStrings;
```
Don't need this

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+
+    public EbbAndFlow() {
+        super(ID, IMG, COST, TYPE, COLOR, RARITY, TARGET);
+        PhaseSystem.subscribe(this);
```
Style-wise I'd say move the subscription below the initialization of the members.

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        // Create an int which equals to your current energy.
```
Kind of superfluous comment (it's also wrong, it's not energy, it's hand size)

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+            this.upgradeName();
+            this.exhaust = false;
+            this.upgradeDamage(2);
+            this.rawDescription = cardStrings.UPGRADE_DESCRIPTION;
```
Unnecessary

strangeglyph replied:
To clarify, setting raw description is unnecessary. InitializeDescription does the work for us already

strangeglyph commented on src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java:
```diff
@@ -0,0 +1,105 @@
+        return new EbbAndFlow();
+    }
+
+    static {
+        cardStrings = CardCrawlGame.languagePack.getCardStrings("EbbAndFlow");
+    }
```
Also unnecessary

WispSS13 committed:
Reset missed PhaseSystem variables to false in Reset()

Fixed missed lines from reset() in PhaseSystem.

General tidying up of EbbAndFlow
